### ISSUE TITLE
brakes angular-cli aot builds

### ISSUE THREAD

**Issue by matulkum** [zefoy/ngx-swiper-wrapper#46]
As soon as I add this module I cannot building production builds with angular cli anymore.

Normally I would call `ng build -prod`. With this module imported into my application the building process will hang somewhere in the middle without giving a error message.

**Comment by sconix** [comment]
At least aot and production builds with angular-cli is working for us. So without error message I don't believe the fault is in thus library. What version of angular-cli you have?

**Comment by matulkum** [comment]
I am using 1.2.7 on macOS. Node version is 8.0.0

Unfortunately I can not get any debug output at all. Even if I build with `ng build -aot --verbose --progress`.  There is nothing useful printed to the command line. It just stops at 10% and utilise my cpu at 100% as soon as I import the Swiper Module.

It is unfortunate, because I really like this module. For now I use "angular2-useful-swiper", which works in that case.

**Comment by matulkum** [comment]
actually I just upgraded to node 8.2.1 and reinstalled ngx-swiper-wrapper from npm. And no it works fine, also with aot.

I have no idea what was the problem before with Node 8.0.0. But it works now.

Sorry for wasting some time. I guess you can close this issue.

Cheers

**Comment by sconix** [comment]
Ok good that you got it resolved. My guess whould have been some version clashes / installation error since there was no clear error message.

We are still using node 6 since it seems to be much less error prone with angular stuff :)

**Comment by guenam** [comment]
I have exactly the same symptoms using node 8.6.0. I'm going to downgrade to node 6.x and note here if it solved it or not.

**Comment by guenam** [comment]
@sconix I spent the last hour trying different combinations and compiling with aot failed as well using node 8.6.0 or node 6.11.4.

Removing ngx-swiper-wrapper package corrected the issue immediately in all cases.

There is no way to tell where the bug is (since compilation hangs without producing any error) but these symptoms appear only when using ngx-swiper-wrapper. No other package I tried in the last months ever produced something similar.

What do you feel about this?

**Comment by sconix** [comment]
Hard to say I have used both node 8 and 7 and never had any problems with pure webpack or angular-cli. And hard to say what could be the issue since I use ngtools to build the library like most of the libraries. And there really is no configuratio for the build process so nothing I could change. So my guess is still some version clash, for example typescript version causes really wierd issues with certain combinations.

**Comment by guenam** [comment]
I just tried using different versions of typescript starting 2.3.x but that didn't help.

I'm an amateur dev so I very quickly reach my limits in such matters, but I have a strong feeling the issue could be related to the UglifyJS library. I had an issue months ago where I had to use a specific version of uglifyjs to get AOT working (but I only just remembered it).

This page [https://github.com/mishoo/UglifyJS2](https://github.com/mishoo/UglifyJS2) states that:
_- uglify-js only supports ECMAScript 5 (ES5).
- Those wishing to minify ES2015+ (ES6+) should use the npm package uglify-es._

The ngx-swiper-wrapper-4.6.6\src\tsconfig.json file mentions "lib": ["dom", "**es2015**"].
And the webpack.config.js refers to the UglifyJS plugin.

Isn't it something to look further there?
